main/steam_tracker.cpp

```diff
@@ -0,0 +1,102 @@
+/*************************************************************************/
+/*  steam_tracker.cpp                                                    */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#if defined(STEAMAPI_ENABLED)
+
+#include "steam_tracker.h"
+
+// https://partner.steamgames.com/doc/sdk/api#initialization_and_shutdown
+
+SteamTracker::SteamTracker() {
+	String path;
+	if (OS::get_singleton()->has_feature("linuxbsd")) {
+		path = OS::get_singleton()->get_executable_path().get_base_dir().plus_file("libsteam_api.so");
+		if (!FileAccess::exists(path)) {
+			path = OS::get_singleton()->get_executable_path().get_base_dir().plus_file("../lib").plus_file("libsteam_api.so");
```

Comment:
Would that work to load the library without setting a `RPATH`? Does Steam automatically add it to `LD_LIBRARY_PATH`?

Reply:
Probably?  I know Steam on Linux has some weird issues sometimes.  We usually tell people to put the .so file in the directory with the editor but even then some need a launcher script to link in that file.  Some people reported they needed the .so file in the project's actual directory (not the editor).  It can vary by distro and, I believe, how Steam it actually installed; where it puts its library files and what versions.